### PR TITLE
build(Webpack): Tidy up configurations and scripts

### LINKED ISSUE #2805: Review development npm/yarn scripts
### Description

Review the following commands in `packages/react-component-library` to make sure they work correctly, use the correct Webpack settings and BABEL_ENV:

- `yarn build:dev`
- `yarn build:ts:dev`
- `yarn dev`

### PR DESCRIPTION
**Pull request by jpveooys**
## Related issue

Resolves #2805

## Overview

This tidies up the Webpack configurations and related npm scripts.

## Reason

To remove broken/unused npm scripts and make the configurations more maintainable.

## Work carried out

- [x] Reduce redundancy in configurations
- [x] Reduce redundancy in npm scripts
- [x] Update/remove dev npm scripts

## Developer notes

Some comments have been left inline.

### CODE REVIEW

File: packages/design-tokens/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   "scripts": {
     "build": "run-s build:dict build:cjs types",
     "build:dict": "node ./build-dictionary.js",
-    "build:cjs": "webpack --mode production --config=webpack/prod.js",
```
Comment (jpveooys) [inline packages/design-tokens/package.json:23]: `--mode production` was redundant as it's specified in the configuration (and is the default also).

File: packages/design-tokens/webpack/common.js [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
-  externals: {
-    'styled-components': {
-      commonjs: 'styled-components',
-      commonjs2: 'styled-components',
-      amd: 'styled-components',
+  output: {
+    filename: 'index.js',
+    path: resolve(__dirname, '../dist/cjs'),
+    library: {
+      type: 'commonjs2',
```
Comment (jpveooys) [inline packages/design-tokens/webpack/common.js:11]: `output.library.type` is recommended over `output.libraryTarget`: https://webpack.js.org/configuration/output/#outputlibrarytarget

File: packages/design-tokens/webpack/common.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
     },
   },
+  externals: {
```
Comment (jpveooys) [inline packages/design-tokens/webpack/common.js:14]: These should only be strings for non-UMD outputs: https://webpack.js.org/configuration/externals/#object

File: packages/react-component-library/package.json [hunk 4]
```diff
@@ -26,14 +26,12 @@
...
   "scripts": {
-    "analyze": "webpack --mode production --config=webpack/analyze.js",
+    "analyze": "webpack --config=webpack/analyze.js",
     "build": "run-s build:es build:cjs types",
-    "build:cjs": "BABEL_ENV=cjs webpack --mode production --config=webpack/prod.js",
+    "build:cjs": "BABEL_ENV=cjs webpack --config=webpack/prod.js",
     "build:es": "BABEL_ENV=es babel src --out-dir \"dist/es\" --extensions \".ts,.tsx\"",
-    "build:dev": "RNDS_LOG_LEVEL=debug run-s build:ts:dev types",
-    "build:ts:dev": "webpack --mode production --config=webpack/dev.js",
+    "build:dev": "BABEL_ENV=cjs RNDS_LOG_LEVEL=debug webpack --config=webpack/dev.js",
```
Comment (jpveooys) [inline packages/react-component-library/package.json]: I've kept `build:dev` for producing a development bundle, and removed the `dev` and `build:ts:dev` scripts.

Comment (m7kvqbe1) [reply packages/react-component-library/package.json]: Should it maybe be renamed to `build:cjs:dev`?

**Review by m7kvqbe1 — APPROVED** [review]
LGTM - just one comment

File: packages/react-component-library/package.json (hunk 4 above)
Comment (jpveooys) [reply packages/react-component-library/package.json]: Sounds good to me!

**Pull request merged**